src/wmtk/operations/tet_mesh/TetSplitWithTags.hpp

```diff
@@ -19,8 +19,10 @@ struct OperationSettings<tet_mesh::TetSplitWithTags>
     MeshAttributeHandle<double> pos_handle;
     long split_vertex_tag_value;
 
+    OperationSettings(const Mesh& m): invariants(m) {}
+
     InvariantCollection invariants;
```

Comment:
to be honest the cleanest thing to do is to make the InvariantCollection held here a shared_ptr. If you don't mind going through that work in every operation then all of this constructor changing stuff changes.

All you need to do is make sure the invariant-collection code everywhere creates a shared_ptr (basic_invariatn_collection or whatnot should return a shared_ptr now), and then you cahnage a bunch of dot notation to arrow notation